src/general/DialogInputHelper.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections.Generic;
+using System.Linq;
+using Godot;
+
+/// <summary>
+///   Helper class for all dialog escapes to be handled here.
+/// </summary>
+public class DialogInputHelper : NodeWithInput
+{
+    private static bool isHelperLoaded;
+
+    /// <summary>
+    ///   If this is not the first helper in Tree, disable this helper.
+    /// </summary>
+    private bool isDuplicate;
+
+    private List<WindowDialog> dialogs = new List<WindowDialog>();
+
+    public override void _EnterTree()
+    {
+        isDuplicate = isHelperLoaded;
+        if (!isHelperLoaded)
+        {
+            isHelperLoaded = true;
+        }
+
+        base._EnterTree();
+    }
+
+    public override void _ExitTree()
+    {
+        if (!isDuplicate)
+        {
+            isHelperLoaded = false;
+        }
+
+        base._ExitTree();
+    }
+
+    /// <summary>
+    ///   When all the nodes are ready search for all dialog windows in the Tree.
+    /// </summary>
+    public override void _Ready()
+    {
+        if (!isDuplicate)
+        {
+            var nodes = new List<Node>();
+            nodes.AddRange(GetTree().Root.GetChildren().OfType<Node>());
+
+            while (nodes.Count != 0)
+            {
+                var currentNode = nodes[0];
+
+                nodes.RemoveAt(0);
+
+                if (currentNode is WindowDialog asDialog)
+                {
+                    // Skip ProcessPanel and CheatMenu and OrganellePopup
+                    if (asDialog.Name == "ProcessPanel"
+                        || asDialog.Name == "CheatMenu"
+                        || asDialog.Name == "OrganellePopup")
+                    {
+                        continue;
+                    }
+
+                    dialogs.Add(asDialog);
+                }
+
+                nodes.AddRange(currentNode.GetChildren().OfType<Node>());
+            }
```

Comment:
I think recursion and linq could clean up and shorten that code a lot

Reply:
I don't prefer deep first search because it includes too much system-handled stack operations. As for linq, I haven't really learned how to use that...